alpenhorn/auto_import.py

```diff
@@ -323,6 +325,52 @@ def get_fileflaginputinfo_keywords_from_h5(path):
     return {"start_time": start_time, "finish_time": finish_time}
 
 
+def get_miscfile_data(path, _file):
+    """Get metadata for a misc-type tarball by reading it.
+
+    Returns a 2-tuple:
+    - first element is a dict with MiscFileInfo data, including _file
+    - second element is a list of dicts of MiscFileList rows
+    """
+
+    serial_number, data_type = di.util.parse_miscfile_name(os.path.basename(path))
+    start_time = None
+    finish_time = None
+    file_list = ()
+
+    with tarfile.open(name=path, mode="r") as f:
+        try:
+            metadata = json.loads(f.extractfile(f.getmember("./METADATA.json")).read())
+            if "start_time" in metadata:
+                start_time = metadata["start_time"]
+            if "finish_time" in metadata:
+                start_time = metadata["finish_time"]
+        except KeyError:
+            pass
+
+        # Generate a list of all regular files
+        file_list = [
+            {
+                "file": _file,
+                "name": member.name,
+                "size_b": member.size,
+                "mtime": member.mtime,
+            }
+            for member in f
+            if member.isfile()
+        ]
+
+    return (
+        {
+            "file": _file,
+            "start_time": start_time,
+            "finish_time": finish_time,
+            "data_type": data_type,
+        },
+        file_list,
```

Comment:
I think all my substantial comments are in the other PR. I think I would like to be able to have an extended description field in the database. That would need to be filled in in this method.